### PR TITLE
[300] Add the diversity confirm page

### PR DESCRIPTION
**Pull request by darokel**
### Context

- https://trello.com/c/OFbecBDU/300-s-confirm-trainee-diversity-details-page

### Changes proposed in this pull request

- Implement the confirm page for the diversity sections
- Sub class main `ConfirmDetails` controller as this section is namespace'd under `diversity`. All the sections within this namespace use the same component. Rather than mix in the logic to deal with those in the main controller (which resolves the confirmation component for each section dynamically), the `Diversity::ConfirmDetailsController` was created to simply render the required component for these specific sections.
- Extract the confirmation component form into a partial

Screenshot 2020-11-05 at 15 43 34

### Guidance to review

Review app: https://dfe-twd-rttd-pr-142.herokuapp.com/

This PR should complete the diversity flow and link all the sections together. Visit the review app and start filling out the diversity section, tweaking the choices and asserting the confirmation page shows the correct information.

### CODE REVIEW

**Comment by darokel** [comment]
[[S] Confirm Trainee Diversity Details Page](https://trello.com/c/OFbecBDU/300-s-confirm-trainee-diversity-details-page)

**Comment by aliuk2012** [comment]
One minor issue with the user journey. once the user has gone through the process and supplied the trainees diversity info then the diversity info link on the overview page should take the user straight back to the confirmation page

**Comment by darokel** [comment]
> One minor issue with the user journey. once the user has gone through the process and supplied the trainees diversity info then the diversity info link on the overview page should take the user straight back to the confirmation page

Yup, I'm leaving that scenario to be dealt with after this is done https://trello.com/c/oozmGsPe/338-m-personal-details-progress-state as it will provide an easy way to check whether the info provided is valid then easily redirect to the confirm page.

File: app/views/trainees/diversity/confirm_details/show.html.erb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+<%= render_component PageTitle::View.new(title: I18n.t("components.confirmation.heading", section_title: trainee_section_key.gsub(/-/, " "))) %>
+
+<%= content_for(:breadcrumbs) do %>
+  <%= render_component GovukComponent::BackLink.new(
+    text: 'Back to overview',
+    href: trainee_path(@trainee)
+  ) %>
+<% end %>
+
+<%= render partial: "trainees/confirm_details/form", locals: { heading: "Confirm diversity information", trainee: @trainee, component: @confirmation_component } %>
```
Comment (defong) [inline app/views/trainees/diversity/confirm_details/show.html.erb:10]: needs newline?

File: spec/support/page_objects/trainees/confirm_details.rb [hunk 2]
```diff
@@ -4,5 +4,9 @@ class ConfirmDetails < PageObjects::Base
       set_url "/trainees/{id}/{section}/confirm"
       element :submit_button, "input[name='commit']"
     end
+
+    class ConfirmDiversityDetails < ConfirmDetails
```
Comment (defong) [inline spec/support/page_objects/trainees/confirm_details.rb:8]: shouldn't it be on its own?

Comment (darokel) [reply spec/support/page_objects/trainees/confirm_details.rb]: yeah i agree, done https://github.com/DFE-Digital/register-trainee-teachers/pull/142/commits/f09be03ac01c0f033bdc4cf56fb709b64bbe222f

File: app/views/trainees/diversity/confirm_details/show.html.erb (hunk 1 above)
Comment (darokel) [reply app/views/trainees/diversity/confirm_details/show.html.erb]: done https://github.com/DFE-Digital/register-trainee-teachers/pull/142/commits/f09be03ac01c0f033bdc4cf56fb709b64bbe222f

**Review by defong — APPROVED** [review]

**Review by JonF1 — APPROVED** [review]

**Pull request merged**